src/services/recipient.test.js

```diff
@@ -1369,51 +1369,6 @@ describe('Recipient DB service', () => {
       expect(result.map((obj) => obj.id)).toEqual([3, 2, 1]); // Sorted by endDate descending
     });
 
-    it('handles undefined activityReports gracefully', () => {
```

Comment:
Why do we remove this test? Do we still handle the null date case?

Reply:
this PR is simply a revert of what was recently introduced in order to resolve this support case

Reply:
in resolving the issue @thewatermethod found, I refactored to handle null/undefined more predictably (which appears to have been the issue all along - unpredictable sort order because of bad values), and reintroduced tests for null/undefined: cb06e9d0416eafddf099420e32412eff6295d856